### DISCUSSION TITLE
Add information about how long is integer and number type

### DISCUSSION THREAD

**Discussion by Robert50364** [json-schema-org/community#504] [Specification]
When I read reference at "Numeric Types" section, I can't find information about  how long is integer and number in bits. Default integer is 32 or 64 bits? I think this information may be helpful.

**Comment by gregsdennis** [comment]
JSON Schema operates on the JSON data model defined by the JSON specification, [RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259).  That spec says that numbers in JSON aren't limited, but notes that, practically, implementations may impose such limits.

Many JSON Schema implementations rely on external JSON parsers, and likely any limitations will be defined there.  You'll need to check with the specific JSON Schema implementation you're using to see what it supports.

For example, in my implementation, _JsonSchema.Net_, I use a .Net `decimal`, which sacrifices a bit of range for improved precision, for floating points and `long` (64-bit) for integers.

**Reply by Robert50364** [reply]
I am using JSON schema to validate yaml files that are used to make a Pod for k8s. In k8s spec (OpenApi3.0) they use "format" key for integer. In one places it is "int32" in another "int64". 

If i understand good. I should add "maximum" key in my JSON schema to limit range of integer to 32 bits if field in OA3.0 limit this?

**Reply by gregsdennis** [reply]
I see.  So they've given you an OpenAPI document, and you're trying to build a schema to validate locally, is that right?  If that's the case, then, yeah, I'd use `maximum` in your schema to place a limit.

`format` isn't validated by default in JSON Schema; it's just an annotation intended for consuming applications to use.  Some implementations may have an option you can use to enable it, but `int32` and `int64` are custom formats defined by OpenAPI, and the implementation may not understand them even with the option enabled.

Regarding which to use, if they use both `int32` and `int64` in different places but _for the same field_, you'd have to go with the smaller one.

**Reply by Robert50364** [reply]
Okay, now I understand everything. Thanks for help and quick advice. 😄
